src/common/node/childProcess.ts

```diff
@@ -55,44 +61,32 @@ export class ChildProcess {
         return this.exec(command, options).outcome.then(stdout => stdout.toString());
     }
 
-    public spawnWithExitHandler(command: string, args?: string[], options: ISpawnOptions = {}): ISpawnResult {
+    public spawn(info: ISpawnInfo, options: ISpawnOptions = {}): ISpawnResult {
```

Comment:
We can do that, but given that we use commands and args 99% of the time, I think we should leave them as normal mandatory parameters. We can put the optional stuff inside an options parameter.